[BUG] Missing necessary license, description, developer information in maven pom

[BUG] Missing necessary license, description, developer information in maven pom

```
[ERROR] 
[ERROR] Nexus Staging Rules Failure Report
[ERROR] ==================================
[ERROR] 
[ERROR] Repository "orgopensearch-1259" failures
[ERROR]   Rule "pom-staging" failures
[ERROR]     * Invalid POM: /org/opensearch/opensearch-geospatial/3.0.0.0-alpha1/opensearch-geospatial-3.0.0.0-alpha1.pom: Project description missing, License information missing, Developer information missing
[ERROR]     * Invalid POM: /org/opensearch/geospatial-client/3.0.0.0-alpha1/geospatial-client-3.0.0.0-alpha1.pom: License information missing, Developer information missing
[ERROR] 
```

1. https://github.com/opensearch-project/geospatial/blob/main/build.gradle#L47 this needs to be set as true
2. Add necessary information just like in Job Scheduler repo: https://github.com/opensearch-project/job-scheduler/blob/main/spi/build.gradle#L127-L157
3. Verify pom file for `opensearch-geospatial` and `geospatial-client` pom in .m2 after running ./gradlew publishNebulaPublicationToMavenLocal


Thanks.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,7 @@ See the [CONTRIBUTING guide](./CONTRIBUTING.md#Changelog) for instructions on ho
 ### Infrastructure
 ### Documentation
 ### Maintenance
+* Persist necessary license and developer information in maven pom ([#732](https://github.com/opensearch-project/geospatial/pull/732))
 ### Refactoring
 
 ## [Unreleased 2.x](https://github.com/opensearch-project/geospatial/compare/2.19...2.x)

--- a/build.gradle
+++ b/build.gradle
@@ -116,7 +116,6 @@ test {
     include '**/*Tests.class'
     systemProperty 'tests.security.manager', 'false'
 }
-
 publishing {
     repositories {
         maven {
@@ -138,22 +137,29 @@ publishing {
                 name = pluginName
                 description = pluginDescription
                 groupId = "org.opensearch.plugin"
-                licenses {
-                    license {
-                        name = "The Apache License, Version 2.0"
-                        url = "http://www.apache.org/licenses/LICENSE-2.0.txt"
-                    }
-                }
-                developers {
-                    developer {
-                        name = "OpenSearch"
-                        url = "https://github.com/opensearch-project/geospatial"
-                    }
-                }
+            }
+        }
+        all {
+            pom {
+                name = pluginName
+                description = pluginDescription
+            }
+            pom.withXml { XmlProvider xml ->
+                Node node = xml.asNode()
+                node.appendNode('inceptionYear', '2021')
+
+                Node license = node.appendNode('licenses').appendNode('license')
+                license.appendNode('name',  "The Apache License, Version 2.0")
+                license.appendNode('url', "http://www.apache.org/licenses/LICENSE-2.0.txt")
+
+                Node developer = node.appendNode('developers').appendNode('developer')
+                developer.appendNode('name', 'OpenSearch')
+                developer.appendNode('url', 'https://github.com/opensearch-project/geospatial')
             }
         }
     }
 }
+
 
 
 configurations {

--- a/client/build.gradle
+++ b/client/build.gradle
@@ -70,22 +70,22 @@ publishing {
         }
     }
     publications {
-        pluginZip(MavenPublication) { publication ->
+        all {
             pom {
                 name = "opensearch-geospatial-client"
                 description = 'OpenSearch Geospatial client'
-                licenses {
-                    license {
-                        name = "The Apache License, Version 2.0"
-                        url = "http://www.apache.org/licenses/LICENSE-2.0.txt"
-                    }
-                }
-                developers {
-                    developer {
-                        name = "OpenSearch"
-                        url = "https://github.com/opensearch-project/geospatial"
-                    }
-                }
+            }
+            pom.withXml { XmlProvider xml ->
+                Node node = xml.asNode()
+                node.appendNode('inceptionYear', '2021')
+
+                Node license = node.appendNode('licenses').appendNode('license')
+                license.appendNode('name',  "The Apache License, Version 2.0")
+                license.appendNode('url', "http://www.apache.org/licenses/LICENSE-2.0.txt")
+
+                Node developer = node.appendNode('developers').appendNode('developer')
+                developer.appendNode('name', 'OpenSearch')
+                developer.appendNode('url', 'https://github.com/opensearch-project/geospatial')
             }
         }
     }

